docs/ug/configFiles.md

```diff
@@ -38,6 +38,7 @@ Ignore Glob List<sup>*+</sup> | The list of file path globs to ignore during ana
 Ignore standalone config | To ignore the standalone config file (if any) in the target repository, enter **`yes`**. If the cell is empty, the standalone config file in the repo (if any) will take precedence over configurations provided in the csv files.
 Ignore Commit List<sup>*+</sup> | The list of commits to ignore during analysis. For accurate results, the commits should be provided with their full hash. Additionally, a range of commits can be specified using the `..` notation, e.g., `abc123..def456` (both inclusive).
 Ignore Authors List<sup>*+</sup> | The list of authors to ignore during analysis. Authors should be specified by their [Git Author Name](#a-note-about-git-author-name).
+Optimize Cloning | Enter **`yes`** to clone the repository using git's shallow cloning functionality. This option can significantly reduce the time taken to clone large repositories. However, the option should ideally be disabled for smaller repositories. Note: this is incompatible with the "Ignore standalone config" option.
```

Comment:
Is this a breaking change? i.e., will this break config files in the old format? In that case, see if you can make it backward compatible.

Reply:
> 
> 
> Is this a breaking change? i.e., will this break config files in the old format? In that case, see if you can make it backward compatible.

As it stands, it is a breaking change, because the CsvParser validates the repo-config.csv file by checking the number of columns. The expected number of columns in the csv file is now 8 rather than 7, so the CsvParser would throw an exception if there are only 7 columns.

We can make it backward compatible by loosening the restriction on the number of columns - we can set a minimum and maximum number of columns in the repo-config.csv file, so we would accept a csv with at least 7 and at most 8 columns. Since the new column is the rightmost, the position of the other columns in the repo-config.csv would all still be the same.

Alternatively, we could instead make CsvParser validate the csv file by checking the column headers. We could make it so that this particular column header is unnecessary, so it just sets a default value of "false". This would also handle cases where a user makes a config csv with the columns in a different order, since the CsvParser currently only cares about the position of the columns (e.g. "File Formats" is assumed to be in column 2) and ignores the headers entirely.

Reply:
> Alternatively, we could instead make CsvParser validate the csv file by checking the column headers. We could make it so that this particular column header is unnecessary, so it just sets a default value of "false". This would also handle cases where a user makes a config csv with the columns in a different order, since the CsvParser currently only cares about the position of the columns (e.g. "File Formats" is assumed to be in column 2) and ignores the headers entirely.

I like this idea. In fact, making some columns optional and not caring about the order of columns would be even better. This may have to be a separate PR. But unless it is done, I'm afraid we can't merge this PR because it will break currently deployed dashboards that use the latest build of RepoSense every time the report is updated.

Reply:
> 
> I like this idea. In fact, making some columns optional and not caring about the order of columns would be even better. This may have to be a separate PR. But unless it is done, I'm afraid we can't merge this PR because it will break currently deployed dashboards that use the latest build of RepoSense every time the report is updated.

Understood, I will make a separate PR with that change. It would also make it easier to add other configuration options going forward without breaking backward compatibility.

Reply:
> 
> 
> > Alternatively, we could instead make CsvParser validate the csv file by checking the column headers. We could make it so that this particular column header is unnecessary, so it just sets a default value of "false". This would also handle cases where a user makes a config csv with the columns in a different order, since the CsvParser currently only cares about the position of the columns (e.g. "File Formats" is assumed to be in column 2) and ignores the headers entirely.
> 
> I like this idea. In fact, making some columns optional and not caring about the order of columns would be even better. This may have to be a separate PR. But unless it is done, I'm afraid we can't merge this PR because it will break currently deployed dashboards that use the latest build of RepoSense every time the report is updated.

Prof, I have implemented this change and ran into an issue while testing. Unfortunately, a bunch of the JUnit tests fail since the config files that are being used for testing themselves do not follow the column header formats exactly.

- AuthorConfigParserTest: test config files have the column header "Author's Email", while the user guide specifies that it should be "Author's Emails"
- GroupConfigParserTest: test config files have the column header "Group label" while the user guide specifies that it should be "Group Name"
- RepoConfigParserTest: test config files have the column header "Ignore Commits List" while the user guide specifies that it should be "Ignore Commit List"

These are relatively small discrepancies. Right now I am using case-insensitive string matching to check the headers, but we can account for these differences quite easily using regex instead. Otherwise, we can just change the headers to match the user guide.

However, my worry is that users may also be using config files that do not match the user guide specification, which would mean that we would still end up breaking backward compatibility.

Reply:
> Prof, I have implemented this change and ran into an issue while testing. Unfortunately, a bunch of the JUnit tests fail since the config files that are being used for testing themselves do not follow the column header formats exactly.
> 
> * AuthorConfigParserTest: test config files have the column header "Author's Email", while the user guide specifies that it should be "Author's Emails"
> * GroupConfigParserTest: test config files have the column header "Group label" while the user guide specifies that it should be "Group Name"
> * RepoConfigParserTest: test config files have the column header "Ignore Commits List" while the user guide specifies that it should be "Ignore Commit List"
> 
> These are relatively small discrepancies. Right now I am using case-insensitive string matching to check the headers, but we can account for these differences quite easily using regex instead. Otherwise, we can just change the headers to match the user guide.
> 
> However, my worry is that users may also be using config files that do not match the user guide specification, which would mean that we would still end up breaking backward compatibility.

Let's make this a separate PR as it is a non-trivial change. Yes, this change itself can be a breaking change, but at least it will make it easier for future changes to be backward compatible.

Reply:
> 
> Let's make this a separate PR as it is a non-trivial change. Yes, this change itself can be a breaking change, but at least it will make it easier for future changes to be backward compatible.

I have made the PR: #1459.